### ISSUE TITLE
How to read onnx model on GPU and set its precision as fp16?

### ISSUE THREAD

**Issue by shiwenloong** [openvinotoolkit/openvino#12448]
I try to read onnx model on GPU. Its default precision if fp32. Here is my code:
```cpp
ov::Core core;
std::shared_ptr<ov::Model> model = core.read_model("a.onnx");
ov::CompiledModel compiled_model = core.compile_model(model, "GPU");
```
I tried to set the model precision like this:
```cpp
ov::Core core;
core.set_property(ov::hint::inference_precision("FP16"))
std::shared_ptr<ov::Model> model = core.read_model("a.onnx");
ov::CompiledModel compiled_model = core.compile_model(model, "GPU");
```
But I got a problem like this:
```bash
Unsupported config key: INFERENCE_PRECISION_HINT
```
Here is SUPPORTED_PROPERTIES on GPU, which is outputted by hello_query_device.
```
GPU
        SUPPORTED_PROPERTIES:
                Immutable: AVAILABLE_DEVICES : 0
                Immutable: RANGE_FOR_ASYNC_INFER_REQUESTS : 1 2 1
                Immutable: RANGE_FOR_STREAMS : 1 2
                Immutable: OPTIMAL_BATCH_SIZE : 1
                Immutable: MAX_BATCH_SIZE : 1
                Immutable: FULL_DEVICE_NAME : Intel(R) UHD Graphics 630 (iGPU)
                Immutable: DEVICE_TYPE : integrated
                Immutable: DEVICE_GOPS : f16 772.8 f32 386.4 i8 386.4 u8 386.4
                Immutable: OPTIMIZATION_CAPABILITIES : FP32 BIN FP16
                Immutable: GPU_DEVICE_TOTAL_MEM_SIZE : 6802485248
                Immutable: GPU_UARCH_VERSION : 9.0.0
                Immutable: GPU_EXECUTION_UNITS_COUNT : 23
                Immutable: GPU_MEMORY_STATISTICS : ""
                Mutable: PERF_COUNT : NO
                Mutable: MODEL_PRIORITY : MEDIUM
                Mutable: GPU_HOST_TASK_PRIORITY : MEDIUM
                Mutable: GPU_QUEUE_PRIORITY : MEDIUM
                Mutable: GPU_QUEUE_THROTTLE : MEDIUM
                Mutable: GPU_ENABLE_LOOP_UNROLLING : YES
                Mutable: CACHE_DIR : ""
                Mutable: PERFORMANCE_HINT : ""
                Mutable: COMPILATION_NUM_THREADS : 6
                Mutable: NUM_STREAMS : 1
                Mutable: PERFORMANCE_HINT_NUM_REQUESTS : 0
                Mutable: DEVICE_ID : 0
```

**Comment by Iffa-Meah** [comment]
@shiwenloong to specifically get your ONNX model to work in FP16 precision, you'll need to use [OpenVINO Model Optimizer](https://docs.openvino.ai/latest/openvino_docs_MO_DG_FP16_Compression.html#doxid-openvino-docs-m-o-d-g-f-p16-compression) to convert the model into IR with FP16 precision.

To ensure that your ONNX is compatible with OepnVINO, run it with benchmark app: benchmark_app -m model.onnx
By default, OpenVINO supports the ONNX native format, therefore if it runs with no error you could proceed to the next step.

Use the conversion command: mo -m model.onnx --data_type FP16

Test the IR with benchmark app: benchmark_app -m model.xml

**Comment by vladimir-paramuzov** [comment]
@shiwenloong inference_precision hint support is implemented in #12526
